tests/lib/cli.js

```diff
@@ -633,6 +633,30 @@ describe("cli", function() {
         });
     });
 
+    describe("when passed --no-inline-config", function() {
+
+        it("should pass allowInlineConfig:false to CLIEngine and ignore inline config when executing on text", function() {
```

Comment:
The description is what you should be doing - making sure the correct options are passed to CLIEngine, but you're testing that via side effects instead of direct inspection, which is fragile. If you look towards the end of this file, you'll see an example of how to test this better.